src/main/scala/io/qbeast/spark/internal/QbeastOptions.scala

```diff
@@ -38,14 +40,44 @@ import org.apache.spark.sql.SparkSession
  *   the application identifier
  * @param userMetadata
  *   user-provided metadata for each CommitInfo
+ * @param mergeSchema
+ *   whether to merge the schema
+ * @param overwriteSchema
+ *   whether to overwrite the schema
  */
 case class QbeastOptions(
     columnsToIndex: Seq[String],
     cubeSize: Int,
     stats: Option[DataFrame],
     txnAppId: Option[String],
     txnVersion: Option[String],
-    userMetadata: Option[String])
+    userMetadata: Option[String],
+    mergeSchema: Option[String],
+    overwriteSchema: Option[String]) {
+
+  def toMap(): Map[String, String] = {
+    val options = Map.newBuilder[String, String]
+    for (txnAppId <- txnAppId; txnVersion <- txnVersion) {
+      options += DeltaOptions.TXN_APP_ID -> txnAppId
+      options += DeltaOptions.TXN_VERSION -> txnVersion
+    }
+    if (userMetadata.nonEmpty) {
+      options += DeltaOptions.USER_METADATA_OPTION -> userMetadata.get
+    }
+    if (mergeSchema.nonEmpty) {
+      options += DeltaOptions.MERGE_SCHEMA_OPTION -> mergeSchema.get
+    }
+    if (overwriteSchema.nonEmpty) {
+      options += DeltaOptions.OVERWRITE_SCHEMA_OPTION -> overwriteSchema.get
```

Comment:
It sounds like codecov is not happy about this line 71. Where do we exercise this `toMap` method in the tests?

Reply:
Fixed on bd53be0 !